lib/senkyoshi/models/content_file.rb

```diff
@@ -26,10 +26,15 @@ def self.correct_linkname(canvas_file)
       canvas_file.file_path.split("/").last
     end
 
-    def canvas_conversion(canvas_file = nil)
-      path = canvas_file ? canvas_file.file_path : @linkname
+    def canvas_conversion(resources, canvas_file = nil)
+      path = if canvas_file
```

Comment:
I'm in the minority here it seems, but I dislike ambiguous assignment statements. (i.e. variable = if...else...end)